### DISCUSSION TITLE
Tips: Using MySqlConnector + MySqlBackup.NET in CentOS (Linux)

### DISCUSSION THREAD

**Discussion by adriancs2** [mysql-net/MySqlConnector#1123] [Show and tell]
I'm one of the main contributor for [MySqlBackup.NET](https://github.com/MySqlBackupNET/MySqlBackup.Net) project, that works on top of MySqlConnector to perform backup & restore of MySQL databases.

The following explained the idea of using MySqlConnector +MySqlBackup.NET in CentOS (Linux).

The idea is to build the project in .NET Core.

We can test this out with a simple Console App.

Starting by creating a .net core linux project:

![01](https://user-images.githubusercontent.com/646615/151099139-d84e64ce-9107-44c7-ba39-1f4103834480.png)

Check the project properties, ensure the project is built with .NET Core framework:

![02](https://user-images.githubusercontent.com/646615/151099158-7c6601dc-e9c4-400b-887a-75bced677f1f.png)

Install Nuget Package of MySqlBackup.NET (with MySqlConnector):

![03](https://user-images.githubusercontent.com/646615/151099167-51975ae3-cafa-4a3b-85cf-201269e58f29.png)

Next, is to publish the project for .NET Core (Linux):

![04](https://user-images.githubusercontent.com/646615/151099175-01f836f8-03df-4366-a6b3-1c8196420982.png)

In above example, the Linux compiled binaries are located at:

```
\bin\Release\netcoreapp3.1\publish\linux-x64
```
Copy the DLL in above folder and paste it to your **CentOS**

At the **CentOS**, make sure that you have installed **Dotnet** runtime by following command:
```
sudo dnf install dotnet-sdk-<version>
```
For example:
```
sudo dnf install dotnet-sdk-3.1
sudo dnf install dotnet-sdk-5.0
```
Assume that you have copied your .NET Core DLL files at the following folder:
```
/home/<username>/testapp
```
Then you can test out your app with following command:
```
cd /home/admin/testapp
dotnet ConsoleApp.dll
```
Here is the sample source code for your referrence:

[ConsoleApp2.zip](https://github.com/MySqlBackupNET/MySqlBackup.Net/files/7938714/ConsoleApp2.zip)

**Comment by bgrainger** [comment]
I may have the exact command-line slightly wrong, but you should be able to run something like `dotnet publish -p:PublishSingleFile=true -c Release --self-contained --arch x64 --os linux` to produce a binary that can run without needing to install the .NET runtime on the target system.

See https://docs.microsoft.com/en-us/dotnet/core/deploying/single-file for more info.

**Comment by adriancs2** [comment]
Publishing self-contained binary can also be done at Visual Studio GUI

![05](https://user-images.githubusercontent.com/646615/151177757-bb930bd4-2504-44b7-af7c-f92f7f0ae3de.png)

Here is the example of how it looks like, there is only 1 single file:

![06](https://user-images.githubusercontent.com/646615/151178601-50f76b67-7153-4e0d-9f4b-9c681889981b.png)

Copy the compiled binary to CentOS

At the terminal, browse to the folder that contains the self-contained binary, for example:

    cd /home/admin/testapp

Next, is to allow execution permission to the file:

    sudo chmod 777 ConsoleApp2

Execute the app:

    ./ConsoleApp2
